### PR TITLE
[^459] Dropdown Select (validation & behavior)

### PR DESCRIPTION
**Pull request by CITguy**
JIRA: fast-follow to SURF-502

* remove `[invalid]`
* add `<hx-select-control>`
  * `[hx-changed]` when 'change' happens on first `<select>` descendant
  * `[hx-touched]` when 'blur' happens on first `<select>` descendant
* `<hx-select>` and `<select>` styled invalid if:
  * `hx-select-control[hx-changed]` and `select:invalid`
  * `hx-select-control[hx-touched]` and `select:invalid`
  * `hx-select-control.hxInvalid` (forced styling by app author)

### LGTM's
- [x] Dev LGTM
- [x] Design LGTM
- [x] Zoom LGTM

### CODE REVIEW

**Comment by HelixUI** [comment]
Deploy preview for *helix-ui* processing.

Building with commit 8fea78108288f685a63aedb59987e04812fd0c10

https://app.netlify.com/sites/helix-ui/deploys/5c74698a8e0d8f0008befb67

File: src/helix-ui/elements/HXSelectControlElement.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        if (ctrl) {
+            ctrl.addEventListener('change', this._onCtrlChange);
+            ctrl.addEventListener('blur', this._onCtrlBlur);
+        }
+    }
+
+    /** @override */
+    $onDisconnect () {
+        let ctrl = this.controlElement;
+
```
Comment (CITguy) [inline src/helix-ui/elements/HXSelectControlElement.js]: remove blank line

File: src/helix-ui/styles/components/dropdown-select.less [hunk 2]
```diff
@@ -1,161 +1,217 @@
...
   }
 
   // Hidden by default, because we can style native <select> in IE.
-  hx-select {
-    -ms-grid-row: 2;
-    grid-area: control;
+  // [002] hx-select-control > hx-select
+  > hx-select {
+    // keep blacklisted for IE
+    z-index: 1;
```
Comment (CITguy) [inline src/helix-ui/styles/components/dropdown-select.less]: may not need `z-index`

File: src/helix-ui/styles/components/dropdown-select.less [hunk 3]
```diff
@@ -1,161 +1,217 @@
...
-  // CSS custom properties to customize appearance.
-  @supports (--modern: true) {
-    hx-select {
-      display: block;
-    }
+  // [002] hx-select-control > select
+  > select {
+    display: block; // whitelisted
+    width: 100%; // prevent overflow
+    z-index: 0;
```
Comment (CITguy) [inline src/helix-ui/styles/components/dropdown-select.less]: may not need `z-index`

File: src/helix-ui/styles/components/dropdown-select.less [hunk 4]
```diff
@@ -1,161 +1,217 @@
...
-    grid-area: control;
-    height: @hxSelect-height;
-    padding-left: 0.75rem; // indent displayed value
+    border: 1px solid @gray-500;
+  }
+
+  // [002] hx-select-control > select
+  > select {
+    background-color: @gray-0;
+    padding-left: 0.75rem; // indent displayed value (undo in modern?)
```
Comment (CITguy) [inline src/helix-ui/styles/components/dropdown-select.less]: does this need reset in modern block?

File: src/helix-ui/styles/components/dropdown-select.less [hunk 5]
```diff
@@ -1,161 +1,217 @@
...
+  }
+
+  .hxInvalid &,
+  &.hxInvalid {
+    > select:enabled {
+      @state: @Select[@Invalid];
+      border: @state[border];
+      color: @state[color];
+
+      // [032] hx-select-control[hx-*] > select:invalid:focus
```
Comment (CITguy) [inline src/helix-ui/styles/components/dropdown-select.less]: correct comment

File: src/helix-ui/styles/components/dropdown-select.less [hunk 6]
```diff
@@ -1,161 +1,217 @@
...
+      &:focus {
+        box-shadow: @focus-glow-invalid;
       }
 
-      // select:disabled ~ hx-select
+      // [023] hx-select-control[hx-*] > select:invalid ~ hx-select
```
Comment (CITguy) [inline src/helix-ui/styles/components/dropdown-select.less]: correct comment

File: src/helix-ui/styles/components/dropdown-select.less (hunk 2 above)
Comment (CITguy) [reply src/helix-ui/styles/components/dropdown-select.less]: will remove

File: src/helix-ui/styles/components/dropdown-select.less (hunk 3 above)
Comment (CITguy) [reply src/helix-ui/styles/components/dropdown-select.less]: will remove

File: docs/components/dropdown-select/index.html [hunk 7]
```diff
@@ -18,37 +20,33 @@ <h2 id="basic-select">Basic Select</h2>
...
       </header>
 
       <div>
-        <div class="hxSelect">
+        <hx-select-control>
           <select
-            id="selDemo"
             :disabled="isDisabled"
             :required="isRequired"
+            @change="onChange"
```
Comment (cathysiller) [inline docs/components/dropdown-select/index.html]: remove event listeners

File: docs/components/dropdown-select/select-demo.js [hunk 8]
```diff
@@ -4,25 +4,47 @@ if (document.getElementById('vue-selectDemo')) {
     new Vue({
         el: '#vue-selectDemo',
         data: {
+            isChanged: false,
             isDisabled: false,
             isRequired: false,
+            isTouched: false,
+        },
+        methods: {
```
Comment (cathysiller) [inline docs/components/dropdown-select/select-demo.js]: remove methods and everything related below

File: src/helix-ui/styles/components/dropdown-select.less (hunk 4 above)
Comment (CITguy) [reply src/helix-ui/styles/components/dropdown-select.less]: nope, stays for modern styles

**Pull request merged**